Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs

```diff
@@ -0,0 +1,1291 @@
+            /// </summary>
+            FlattenAuto,
+        }
+
+        private enum HandleType
+        {
+            None = 0,
+            Rotation,
+            Scale
+        }
+
+        private enum WireframeType
+        {
+            Cubic = 0,
+            Cylindrical
+        }
+
+        private enum CardinalAxisType
+        {
+            X = 0,
+            Y,
+            Z
+        }
+
+        private enum BoundsCalculationMethod
+        {
+            Collider = 0,
+            Colliders,
+            Renderers,
+            MeshFilters
+        }
+
+        private enum HandleMoveType
+        {
+            Ray = 0,
+            Point
+        }
+
+        #endregion Enums
+
+        #region Serialized Fields
+
+        [Header("Bounds Calculation")]
+        [Tooltip("For complex objects, automatic bounds calculation may not behave as expected. Use an existing Box Collider (even on a child object) to manually determine bounds of Bounding Box.")]
+        [SerializeField]
+        private BoxCollider boxColliderToUse = null;
+
+        [Header("Behavior")]
+        [SerializeField]
+        private bool activateOnStart = false;
+
+        [SerializeField]
+        private float scaleMaximum = 2.0f;
+
+        [SerializeField]
+        private float scaleMinimum = 0.2f;
+
+        [Header("Wireframe")]
+        [SerializeField]
+        private bool wireframeOnly = false;
+
+        public bool WireframeOnly
+        {
+            get { return wireframeOnly; }
+            set
+            {
+                if (wireframeOnly != value)
+                {
+                    wireframeOnly = value;
+                    ResetHandleVisibility();
+                }
+            }
+        }
+
+        [SerializeField]
+        private Vector3 wireframePadding = Vector3.zero;
+
+        [SerializeField]
+        private FlattenModeType flattenAxis = FlattenModeType.DoNotFlatten;
+
+        [SerializeField]
+        private WireframeType wireframeShape = WireframeType.Cubic;
+
+        [SerializeField]
+        private Material wireframeMaterial;
+
+        [Header("Handles")]
+        [Tooltip("Default materials will be created for Handles and Wireframe if none is specified.")]
+        [SerializeField]
+        private Material handleMaterial;
+
+        [SerializeField]
+        private Material handleGrabbedMaterial;
+
+        [SerializeField]
+        private bool showScaleHandles = true;
+
+        public bool ShowScaleHandles
+        {
+            get
+            {
+                return showScaleHandles;
+            }
+            set
+            {
+                if (showScaleHandles != value)
+                {
+                    showScaleHandles = value;
+                    ResetHandleVisibility();
+                }
+            }
+        }
+
+        [SerializeField]
+        private bool showRotateHandles = true;
+
+        public bool ShowRotateHandles
```

Comment:
please add summary tag for api documentation generation

Reply:
👍 done